src/dag.rs

```diff
@@ -125,4 +125,240 @@ where
 }
 
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::*;
+    use petgraph::graph::DiGraph;
+    use std::collections::HashSet;
```

Comment:
@coderabbitai Please import `use pretty_assertions::assert_eq;` so we have pretty results of unit tests.